mpsp/src/parser.py

```diff
@@ -0,0 +1,149 @@
+import pandas as pd
+from pandas_ods_reader import read_ods
+from openpyxl import load_workbook
+import math
+import sys
+import os
+import pathlib
+
+cwd = os.getcwd()
+
+def read_data(path):
+    if('ods' in path):
+        try:
+            data = read_ods(cwd + path, 1)
+        except Exception as excep:
+            sys.stderr('Não foi possível fazer a leitura do arquivo: ' + path +'e o seguinte error foi gerado:' + str(excep))
+            os._exit(1)
+    else:
+        try:
+             data = pd.read_excel(cwd + path, engine='openpyxl')
+        except Exception as excep:
+            sys.stderr('Não foi possível fazer a leitura do arquivo: ' + path +'e o seguinte error foi gerado:' + str(excep))
+            os._exit(1)
+
+    return data
+
+# Strange way to check nan. Only I managed to make work
+# Source: https://stackoverflow.com/a/944712/5822594
+def isNaN(string):
+    return string != string
+
+def get_begin_row(rows):
+    begin_row = 0
+
+    # We need to continue interate until wee a value that is not
+    # None or NaN. That happen due to the spreadsheet formatting.
+    while rows[begin_row][0] == None or isNaN(rows[begin_row][0]):
+        begin_row += 1
+
+    return begin_row
+
+
+def get_end_row(rows, begin_row, file_name):
+    end_row = begin_row 
+    array_len = len(rows)
+   
+    #There are variations at the end of the worksheets. Different conditions are needed
+    if 'Pensionistas' in file_name:
+        end_row = array_len
+    elif(type(rows[array_len-1][0]) == float):
+        end_row = array_len
+    else: 
+        # Then keep moving until find a different type of float.  
+        while type(rows[end_row][0]) == float:
+            end_row += 1
+       
+    end_row -= 1
+    return end_row 
+
+def type_employee(fn):
+    if 'Membros' in fn:
+        return 'membro'
+    if 'Servidores' in fn:
+        return 'servidor'
+    if 'Pensionistas' in fn:
+        return 'pensionista'
+    if 'Colaboradores' in fn:
+        return 'colaborador'
+    raise ValueError('Tipo de inválido de funcionário público: ' + fn)
+
+# Adjust existing spreadsheet variations
+def format_value(element):
+    if(element == None):
+        return 0.0
+    if(type(element) == str and '-' in element):
+        return 0.0
+    return element
+
+# Parser for the months from July 2019 to November 2020 
+def parse_employees(file_name):
+    rows = read_data(file_name).to_numpy()
+    begin_row = get_begin_row(rows)
+    end_row = get_end_row(rows, begin_row, file_name)
+
+    typeE = type_employee(file_name)
+    activeE = 'inativos' not in file_name 
+    employees = {}
+    curr_row = 0
+    for row in rows:
+        if curr_row < begin_row:
+            curr_row += 1
+            continue
+    
+        employees[str(int(row[0]))] = {
+            'reg': str(int(row[0])),
+            'name': row[1],
```

Comment:
Precisamos de mais clareza nessas células básicas. Minha sugestão é ter algo como:

```python
mat = row[0]
name = row[1]

# seguir atribuindo as células a nomes. quando os nomes das variáveis não forem claros, colocar como comentário o cabeçalho das colunas.
```

Depois disso você faz a atribuição e os somatórios. Creio que isso vai tornar o código muito mais claro e fácil de manter.

Reply:
Não sei se entendi correto, alterei somente no parser principal. Se não for do jeito que tá pode falar :)